src/backend/distributed/operations/auto_shard_split.c

```diff
@@ -0,0 +1,251 @@
+#include "postgres.h"
+#include "libpq-fe.h"
+#include "executor/spi.h"
+#include "distributed/lock_graph.h"
+#include "distributed/coordinator_protocol.h"
+#include "distributed/metadata_cache.h"
+#include "distributed/metadata_utility.h"
+#include "distributed/multi_logical_replication.h"
+#include "distributed/multi_server_executor.h"
+#include "distributed/pg_dist_rebalance_strategy.h"
+#include "distributed/pg_dist_shard.h"
+#include "distributed/reference_table_utils.h"
+#include "distributed/remote_commands.h"
+#include "distributed/resource_lock.h"
+#include "distributed/tuplestore.h"
+#include "distributed/utils/array_type.h"
+#include "distributed/worker_protocol.h"
+#include "nodes/pg_list.h"
+#include "postmaster/postmaster.h"
+
+
+PG_FUNCTION_INFO_V1(citus_auto_shard_split_start); 
+#define MAX_SHARD_SIZE 20000
+
+typedef struct ShardInfoData
+{
+    int64 shardsize;
+    int64 shardminvalue;
+    int64 shardmaxvalue;
+    int64 shardid ;
+    int64 nodeid ;
+    char *tablename;
+    char *distributionColumn;
+    char *datatype;
+
+}ShardInfoData;
+typedef ShardInfoData *shardInfo;
```

Comment:
shardInfo -> ShardInfo, to make it more clear that this is a type

Reply:
Implemented